lens/src/mappings/index.ts

```diff
@@ -0,0 +1,187 @@
+    event.logIndex,
+    ADDRESS_ZERO,
+    to.toHexString(),
+    profileId.toString(),
+    LENSHUB_ADDRESS.toHexString(),
+    to.toHexString(),
+    new Array<airstack.social.AirExtraData>(),
+    event.params.handle,
+    new Array<airstack.social.AirExtraData>(),
+    BIG_INT_ZERO,
+  )
+}
+export function handleTransfer(event: Transfer): void {
+  let txHash = event.transaction.hash
+  let transferEntity = TransferEntity.load(event.params.tokenId.toString())
+  if (transferEntity == null) {
+    transferEntity = new TransferEntity(event.params.tokenId.toString())
+    transferEntity.hashes = [txHash.toHexString()]
+    transferEntity.fromTo = [
+      event.params.from
+        .toHexString()
+        .concat('-')
+        .concat(event.params.to.toHexString()),
+    ]
+    transferEntity.tokenId = event.params.tokenId
+    transferEntity.count = BIGINT_ONE
+  } else {
+    let hashArr = transferEntity.hashes
+    if (hashArr == null) {
+      throw new Error('')
+    } else {
+      hashArr.push(txHash.toHexString())
+    }
+    transferEntity.hashes = hashArr
+    let fromTo = transferEntity.fromTo
+    if (fromTo == null) {
+      throw new Error('')
+    } else {
+      fromTo.push(
+        event.params.from
+          .toHexString()
+          .concat('-')
+          .concat(event.params.to.toHexString()),
+      )
+    }
+    transferEntity.fromTo = fromTo
+    transferEntity.count = transferEntity.count.plus(BIGINT_ONE)
+  }
+  transferEntity.save()
+  if (event.params.from.toHexString() != ADDRESS_ZERO) {
+    // changing default Profile if this profileId is previous default
+    let profileId = getDefaultProfileIdfromUser(event.params.from)
+    if (profileId == event.params.tokenId) {
+      log.debug('default profile reset ,hash {}', [txHash.toHexString()])
+      airstack.social.trackSocialUserDefaultProfileChange(
+        event.block,
+        txHash.toHexString(),
+        event.logIndex,
+        event.transaction.from.toHexString(),
+        event.address.toHexString(),
+        '',
+        LENSHUB_ADDRESS.toHexString(),
+        event.params.from.toHexString(),
+      )
+      // removing from mapping
+      createOrUpdateUserToDefaultProfileIdMap(event.params.from, BIG_INT_ZERO)
+    }
+    airstack.social.trackSocialProfileOwnershipChangeTransaction(
+      event.block,
+      txHash.toHexString(),
+      event.logIndex,
+      event.params.from.toHexString(),
+      event.params.to.toHexString(),
+      event.params.tokenId.toString(),
+      LENSHUB_ADDRESS.toHexString(),
+      event.params.to.toHexString(),
+    )
+  }
+}
+export function handleDefaultProfileSet(event: DefaultProfileSet): void {
+  let txHash = event.transaction.hash
+  let defaultProfileSet = DefaultProfileSetEntity.load(
+    txHash
+      .toHexString()
+      .concat('-')
+      .concat(event.logIndex.toString()),
+  )
+  if (defaultProfileSet != null) {
+    log.error('defaultProfileSet already exists,hash {}', [txHash.toHexString()])
```

Comment:
why are we throwing error here? can someone not set one profile as defualt and then change it to another profile of theirs?

Reply:
just for logging - removed